probe/src/probe/publisher/publisher.cpp

```diff
@@ -27,6 +28,14 @@ publisher::~publisher() {
     delete m_client_event_map;
 }
 
+bool filterSwitch(char *val, int threshold){
+    int len = strlen(val), num = 0;
+    for(int i = 0;i < len;i++){
+        num *= 10;
+        num += val[i] - '0';
+    }
```

Comment:
I don't think it is a good idea to write your own function to convert `char*` to `int` since there is a simple function `atoi` in the standard library which can simplify your codes and ensure the logic is correct. I'm really concerned about these non-test codes, especially when it is wrote in C++.